packages/grafana-data/src/types/dataFrame.ts

```diff
@@ -98,6 +99,8 @@ export interface FieldConfig<TOptions = any> {
   // The behavior when clicking on a result
   links?: DataLink[];
 
+  actions?: Action[];
```

Comment:
🤔  if we are adding this to FieldConfig, we also need a plan for what it looks like in the [go SDK](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/data/field_config.go#L11)

The Action type is not part of this PR, so a bit hard to comment here... as structured it would be non-obvious how to have good go-struct support.  How bad would it be to change the pattern to something like:  https://github.com/grafana/grafana/blob/v10.1.1/pkg/services/live/pipeline/config.go#L37

```
export type ActionConfig = {
  type: ActionType;
  title: string; 

  // NOTE the key matches the options type
  fetch?: FetchOptions;
}
```

Reply:
i think that's fine.

we'd have to tweak these bits to work with the recently-merged Canvas Actions PR: https://github.com/grafana/grafana/pull/90677/files#diff-d2dcf441507cf28f433d0b9482bd9fe12d889894654314f88225b76c8b293bd6R6-R10

@ryantxu do you want to make a PR that makes these changes to the existing code and introduces the SDK side of it, then we can rebase this PR on that?